jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,42 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+
+    directory_size = self.get_size_of_directory()
+    new_file_size = 0
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+      new_file_size = self.creating_a_file(tmpdir, key, value)
+
+    if(new_file_size + directory_size > self._max_cache_size):
+      last_time = 0
+      file_to_delete = ""
+
+      for files in os.listdir(self._path):
+        file_to_inspect = os.path.join(self._path, files)
+
+        if os.path.getmtime(file_to_inspect)  > last_time:
+          last_time = os.path.getmtime(file_to_inspect)
+          file_to_delete = file_to_inspect
+
+      os.remove(file_to_delete)
```

Comment:
You might have to remove multiple files if `value` is larger than any existing file. You should add a test case for this, and fix this function to handle this case (I think right now it will only remove a maximum of 1 file).

Reply:
I added a while loop here but is there a more efficient way to do this?